app/assets/javascripts/discourse/app/components/bookmark-actions-dropdown.js

```diff
@@ -11,7 +12,9 @@ export default DropdownSelectBoxComponent.extend({
     showFullTitle: true,
   },
 
-  content: computed(() => {
+  @discourseComputed("bookmark")
+  content(bookmark) {
+    let pinVerb = bookmark.pinned ? "unpin" : "pin";
```

Comment:
what about we add a property to bookmark model with this logic?

```
bookmark.toHumanState() // or similar you will probably find something better than me
```

Reply:
I am not sure what you mean, do you mean this pin verb logic specifically? I could add something like `bookmark.pinAction()`?

Reply:
Yes I feel like this whole line shouldnt be here but in model